sockeye/inference.py

```diff
@@ -630,45 +635,86 @@ def make_input(sentence_id: int, sentence: str) -> TranslatorInput:
         tokens = list(data_io.get_tokens(sentence))
         return TranslatorInput(id=sentence_id, sentence=sentence.rstrip(), tokens=tokens)
 
-    def translate(self, trans_input: TranslatorInput) -> TranslatorOutput:
+    def translate(self, trans_inputs: List[TranslatorInput]) -> List[TranslatorOutput]:
         """
-        Translates a TranslatorInput and returns a TranslatorOutput
+        Batch-translates a list of TranslatorInputs, returns a list of TranslatorOutputs.
+        Splits oversized sentences to sentence chunks of size less than max_input_length.
 
-        :param trans_input: TranslatorInput as returned by make_input().
-        :return: translation result.
+        :param trans_input: List of TranslatorInputs as returned by make_input().
+        :return: List of translation results.
         """
-        if not trans_input.tokens:
-            return TranslatorOutput(id=trans_input.id,
-                                    translation="",
-                                    tokens=[""],
-                                    attention_matrix=np.asarray([[0]]),
-                                    score=-np.inf)
+        # keep input ids to be able to concatenate translated sentence chunks back
+        chunk_markers = []
```

Comment:
you know my opinion about sorting here. Let's leave it out for now, but reconsider later.